tensorrt_llm/_torch/models/modeling_deepseekv3.py

```diff
@@ -134,585 +134,873 @@ def moe_reduce_add_shared_output(routed_output, shared_output):
     return shared_output + routed_output
 
 
-class DeepseekV3MTPHead(nn.Module):
```

Comment:
There are lots of diffs in this file. Can we revert the weight loading refactor and any other moved code for now? It's a bit hard to review what's actually changed right now. I assume it's mostly just moving code around, but it would be best practice to do refactoring in another PR since that's orthogonal to what we're trying to do here